test/thumbnailer_test.cc

```diff
@@ -102,110 +114,38 @@ class WebPTestGenerator {
-                                                           WebPDataDelete);
-  WebPDataInit(webp_data.get());
-  ASSERT_TRUE(WebPTestGenerator(pic_count, transparency, use_randomized)
-                  .GenerateWebPData(webp_data.get(), equal_psnr));
-  EXPECT_LE(webp_data->size, kDefaultBudget);
-  EXPECT_GT(webp_data->size, 0);
-}
-
-TEST(ThumbnailerTest, NoisyImageTransparent) {
-  const int pic_count = 10;
-  const uint8_t transparency = 0xaf;
-  const bool use_randomized = true;
-  const bool equal_psnr = false;
+  libwebp::Thumbnailer thumbnailer = libwebp::Thumbnailer();
+  auto pics =
+      WebPTestGenerator(pic_count, transparency, use_randomized).GeneratePics();
+  for (int i = 0; i < pic_count; ++i) {
+    ASSERT_EQ(thumbnailer.AddFrame(*pics[i], i * 500),
+              libwebp::Thumbnailer::kOk);
+  }
   std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
                                                            WebPDataDelete);
   WebPDataInit(webp_data.get());
-  ASSERT_TRUE(WebPTestGenerator(pic_count, transparency, use_randomized)
-                  .GenerateWebPData(webp_data.get(), equal_psnr));
-  EXPECT_LE(webp_data->size, kDefaultBudget);
-  EXPECT_GT(webp_data->size, 0);
-}
 
-TEST(ThumbnailerPSNRTest, NoisyImageSolid) {
-  const int pic_count = 10;
-  const uint8_t transparency = 0xff;
-  const bool use_randomized = true;
-  const bool equal_psnr = true;
-  std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
-                                                           WebPDataDelete);
-  WebPDataInit(webp_data.get());
-  WebPDataInit(webp_data.get());
-  ASSERT_TRUE(WebPTestGenerator(pic_count, transparency, use_randomized)
-                  .GenerateWebPData(webp_data.get(), equal_psnr));
+  ASSERT_EQ(thumbnailer.GenerateAnimation(webp_data.get()),
+            libwebp::Thumbnailer::kOk);
   EXPECT_LE(webp_data->size, kDefaultBudget);
   EXPECT_GT(webp_data->size, 0);
 }
 
-TEST(ThumbnailerPSNRTest, NoisyImageTransparent) {
-  const int pic_count = 10;
-  const uint8_t transparency = 0xaf;
-  const bool use_randomized = true;
-  const bool equal_psnr = true;
-  std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
-                                                           WebPDataDelete);
-  WebPDataInit(webp_data.get());
-  ASSERT_TRUE(WebPTestGenerator(pic_count, transparency, use_randomized)
-                  .GenerateWebPData(webp_data.get(), equal_psnr));
-  EXPECT_LE(webp_data->size, kDefaultBudget);
-  EXPECT_GT(webp_data->size, 0);
-}
+INSTANTIATE_TEST_CASE_P(ThumbnailerTest, GenerateAnimationTest,
+                        ::testing::Values(std::make_tuple(10, 0xff, false),
+                                          std::make_tuple(10, 0xaf, false),
+                                          std::make_tuple(10, 0xff, true),
+                                          std::make_tuple(10, 0xaf, true)));
```

Comment:
Good, it is much shorter!
You can also use `::testing::Combine(::testing::Values(10), ::testing::Values(0xff, 0xaf), ::testing::Values(false, true))` or similar to generate the same test cases.